File: query.py
import pymongo
from pymongo import MongoClient
client = MongoClient('localhost', 27017)
collection = client['idea']['geo_tweets']
cursor=collection.find({"lang":"en"})
#print("tweet_id\tuser_id\tdate\ttext")
for data in cursor:
	try:
		tweet_id = data["id"]
		user_id = data["user"]["id"]
		date = data["created_at"]
		text = data["text"]
		lon, lat = data["coordinates"]["coordinates"]

		print("%d\t%s\t%s\t%f\t%f" % (user_id,text,date,lon,lat))
	except:
		continue

File: query.rb
require 'json'
require 'mongo'
include Mongo

 mongo_client = MongoClient.new("localhost", 27017)
 coll = mongo_client['idea']['geo_tweets']
 #mongo_client.database_info.each { |info| puts info.inspect }

find_twitters = Proc.new{ |lang|
	w = File.open "#{lang}_data.csv","w"
	#w.write "text\n"
	incorrect = 0
	coll.find("lang" => lang).each do |data|
		begin
			text = data["text"]
			#coordinates =  data["coordinates"]["coordinates"]
			#lon,lat = coordinates
			#date = data["created_at"]
			w.write "#{text}\n"
		rescue
			incorrect +=1
		end

	end
	puts incorrect
	}

["zh"].each do |lang|
	find_twitters.call lang
end
